Only use zypp::*_ptr instead of boost::*_ptr

gruenich opened the pull request:
Preparation for a future switch of `boost::*_ptr` to C++11 `std::*_ptr`. This needs care and a transition as software using libzypp needs adaption.
*_ptr = shared_ptr, scoped_ptr, weak_ptr, intrusive_ptr

mlandres reviewed (changes requested):
- Please keep in mind that the branch feeds released products. We must stay binary compatible! We could easily switch to std:: but we will not without an urgent need. 
- Please don't change the .changes file. This will create unnecessary merge conflicts. Our submission scripts create a new .changes template by collecting the commit messages added since the last release. So it is sufficient to have good texts there.
- Regarding the code, I'd prefer not to use zypp:: qualified type names inside the zypp namespace.

gruenich commented on zypp/ZYpp.h:
```diff
@@ -59,9 +59,8 @@ namespace zypp
     friend std::ostream & operator<<( std::ostream & str, const ZYpp & obj );
 
   public:
-    // can't get swig working if shared_ptr is without namespace here
```
1. Binary compatibility is not touched. This pull is a first preparational step for a future migration.
2. Removed the changes from .changes.
3. My though was to make it explicit, to avoid confusion. It is not uncommon that code contains `using std::shared_ptr` somewhere. I do not insists, if it is a deal-breaker for you.

gruenich commented:
1. Binary compatibility is not touched. This pull is a first preparational step for a future migration.
2. Removed the changes from .changes.
3. My though was to make it explicit, to avoid confusion. It is not uncommon that code contains `using std::shared_ptr` somewhere. I do not insists, if it is a deal-breaker for you.

mlandres commented:
Regarding 1. I meant that changing from boot:: to std:: _would break_ it and that's why we will not change it (without need). So the announcement of a _future migration to std::_ has no real foundation. But we like to be able to.

Regarding 3. You are not wrong and there might be no killer pro or con. Code outside the zypp namespace may use whatever using directive they want. `using std::shared_ptr` there does not change the meaning of `shared_ptr` inside the zypp namespace. I'd prefer that zypp types inside the namespace are not zypp:: qualified without need. This is how (the majority of) the code is written. In case of doubt I try to follow the surrounding code, otherwise people ask why it ts different.

gruenich commented:
1. I added a "potential" in front of "future change" to not implicate any future plan. I still think with a compile time switch one could preserve the binary compatibility for older distributions and use modern C++ in all other cases. I will create a draft in the coming days to demonstrate what I want to achieve.

3. Removed all `zypp::` in front of `shared_ptr`.
